### PR TITLE
Introduce ProfilingSessionEventSource/ProfilingSessionTraceProcessor

### PR DESCRIPTION
**Pull request by pharring**
Adds a ProfilingSessionEventSource and ProfilingSessionTraceProcessor to Azure.Monitor.OpenTelemetry.AspNetCore.

The ProfilingSessionEventSource listens (via command arguments passed into OnEventCommand) for a profiling session ID.
When the profiling session ID is set, the ProfilingSessionTraceProcessor adds a tag to each Activity so that the session ID can be exported as a custom attribute on each span.

The idea is that profilers (e.g. dotnet-trace, dotnet-monitor, PerfView, WPR) can communicate a unique session ID to the application or applications being profiled. They do so by enabling the Azure-Monitor-ProfilingSession EventSource, passing in the Session ID as a filter parameter. This session ID may then be used by the profiler to tag or name the resulting trace file (.nettrace, .ETL, etc.) and/or to archive for later retrieval.

Unit tests are included with this PR.

### CODE REVIEW

File: sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/src/Vendoring/Shared/ProfilingSessionTraceProcessor.cs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    /// if a profiling session is active. This enables correlation between
+    /// traces and profiler artifacts.
+    /// </summary>
+    internal sealed class ProfilingSessionTraceProcessor : BaseProcessor<Activity>
+    {
+        /// <summary>
+        /// Name of the Tag (and subsequent OpenTelemetry attribute) that
+        /// records the profiling session ID.
+        /// </summary>
+        internal const string TagName = "ds.profiling.session";
```
Comment (pharring) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/src/Vendoring/Shared/ProfilingSessionTraceProcessor.cs:20]: @rajkumar-rangaraj Need your opinion on the right name for the tag/attribute. Is there a convention?

**Review by jsquire — DISMISSED** [review]
@pharring :

Your account needs adjustments before this PR can move forward.  As an internal contributor, you'll need to have Microsoft and Azure organization memberships set to public on your profile and request write access to the repository.

Please review the Azure SDK onboarding documentation and use the associated Teams channel for support/discussion.
- [Azure SDK onboarding](https://eng.ms/docs/products/azure-developer-experience/onboard/requirements) _(Microsoft internal)_
- [Azure SDK onboarding assistance](https://teams.microsoft.com/l/channel/19%3ade3fce22c2994be18cac50502c55f717%40thread.skype/Azure%2520SDK%2520Onboarding?groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47) _(Microsoft internal)_

Please also be sure to add yourself to [CODEOWNERS](https://github.com/Azure/azure-sdk-for-net/blob/main/.github/CODEOWNERS) for this library, if you will be maintaining it going forward.

**Comment by TimothyMothra** [comment]
Hi Paul!

You put your code in the "Vendoring" subdirectory. This is used for code we're copying from the opentelemetry-dotnet repo. I don't want your code to get accidentally deleted, so can you please move it to another directory? The convention we're using in the other projects is an "Internals" subdirectory. I recommend using an "Internals/Profiling" directory.

**Comment by pharring** [comment]
@TimothyMothra I moved the two new files as you suggested. Please take another look.

File: sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/src/Internals/Profiling/ProfilingSessionEventSource.cs [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    /// EventPipe. Profilers should use the command to notify applications of
+    /// a profiling session ID. The session ID will be included as an
+    /// attribute on any application telemetry generated while profiling is
+    /// active. This session ID may be later used to correlate telemetry with
+    /// the resulting profiler artifact.
+    /// The GUID of this event source is 15ec0b5c-cb74-5fec-5d64-609c0a49ff31
+    /// </summary>
+    /// <remarks>
+    /// PerfView instructions:
+    /// <code>PerfView.exe collect -MaxCollectSec:300 -NoGui /Providers=/Providers:"*Azure-Monitor-ProfilingSession:::SessionId=SESSIONID"</code>
```
Comment (rajkumar-rangaraj) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/src/Internals/Profiling/ProfilingSessionEventSource.cs:23]: nit:
```suggestion
    /// <code>PerfView.exe collect -MaxCollectSec:300 -NoGui /Providers=*Azure-Monitor-ProfilingSession:::SessionId=SESSIONID</code>
```

File: sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/src/Vendoring/Shared/ProfilingSessionTraceProcessor.cs (hunk 1 above)
Comment (rajkumar-rangaraj) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/src/Vendoring/Shared/ProfilingSessionTraceProcessor.cs]: There are no OpenTelemetry semantics for profiling. Think this TagName is good.  Is there a reason to prefix `ds`?

File: sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/src/Internals/Profiling/ProfilingSessionEventSource.cs [hunk 3]
```diff
@@ -0,0 +1,88 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+#nullable enable
+
+using System.Diagnostics.Tracing;
+
+namespace Azure.Monitor.OpenTelemetry
```
Comment (rajkumar-rangaraj) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/src/Internals/Profiling/ProfilingSessionEventSource.cs:7]: Please use the namespace `Azure.Monitor.OpenTelemetry.AspNetCore.Internals.Profiling`, as we match the folder name to the namespace.

Comment (rajkumar-rangaraj) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/src/Internals/Profiling/ProfilingSessionEventSource.cs:7]: Please follow the same pattern for any other files inside this folder.

File: sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/src/Vendoring/Shared/ProfilingSessionTraceProcessor.cs (hunk 1 above)
Comment (rajkumar-rangaraj) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/src/Vendoring/Shared/ProfilingSessionTraceProcessor.cs]: @cijothomas Do you have any recommendations here for the tag name?

**Review by rajkumar-rangaraj — APPROVED** [review]
LGTM, left some minor comments.

File: sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/src/Vendoring/Shared/ProfilingSessionTraceProcessor.cs (hunk 1 above)
Comment (pharring) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/src/Vendoring/Shared/ProfilingSessionTraceProcessor.cs]: "ds" is for "Diagnostic Services", but that's an internal team name leaking through. How about just "profiling.session" ?

Comment (rajkumar-rangaraj) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/src/Vendoring/Shared/ProfilingSessionTraceProcessor.cs]: `profiling.session` should be good to use.

**Comment by pharring** [comment]
@rajkumar-rangaraj build failures seem unrelated. Otherwise, is this ready to merge?

**Comment by rajkumar-rangaraj** [comment]
> @rajkumar-rangaraj build failures seem unrelated. Otherwise, is this ready to merge?

Yes, we could merge it. CI has some issues, I reran the tests to fix it.

**Pull request merged**